src/monexamine.cpp

```diff
@@ -611,24 +602,14 @@ void monexamine::remove_harness( monster &z )
 void monexamine::remove_armor( monster &z )
 {
     std::string pet_name = z.get_name();
-    bool found_armor = false;
-    int pos = 0;
-    for( auto &it : z.inv ) {
-        if( it.is_pet_armor( true ) ) {
-            found_armor = true;
-            it.erase_var( "pet_armor" );
-            g->m.add_item_or_charges( z.pos(), it );
-            //~ %1$s: armor name, %2$s: pet name
-            add_msg( m_info, pgettext( "pet armor", "You remove the %1$s from %2$s." ), it.tname( 1 ),
-                     pet_name );
-            z.inv.erase( z.inv.begin() + pos );
-            g->u.moves -= 200;
-            break;
-        } else {
-            pos += 1;
-        }
-    }
-    if( !found_armor ) {
+    if( z.armor_item ) {
+        z.armor_item->erase_var( "pet_armor" );
+        g->m.add_item_or_charges( z.pos(), z.armor_item.value() );
```

Comment:
`cata::optional` objects are usually accessed via `operator*`.
```suggestion
        g->m.add_item_or_charges( z.pos(), *z.armor_item );
```
It's shorter.

It's also more consistent with your usage of it in `monster::get_carried_weight` (where you use the `operator->` instead of `value`).